drivers/clock_control/clock_stm32_mco.c

```diff
@@ -0,0 +1,136 @@
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_MSIK
+#elif DT_PROP(DT_NODELABEL(mco1), source) == STM32_SRC_MSIS
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_MSIS
+#elif DT_PROP(DT_NODELABEL(mco1), source) == STM32_SRC_HSI
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_HSI
+#elif DT_PROP(DT_NODELABEL(mco1), source) == STM32_SRC_HSI16
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_HSI
+#elif DT_PROP(DT_NODELABEL(mco1), source) == STM32_SRC_HSI48
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_HSI48
+#elif DT_PROP(DT_NODELABEL(mco1), source) == STM32_SRC_PLLCLK
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_PLLCLK
+#elif DT_PROP(DT_NODELABEL(mco1), source) == STM32_SRC_PLLQCLK
+	#if (CONFIG_SOC_SERIES_STM32G0X || CONFIG_SOC_SERIES_STM32WLX)
+		#define MCO1_SOURCE	LL_RCC_MCO1SOURCE_PLLQCLK
+	#elif (CONFIG_SOC_SERIES_STM32H5X || \
+		 CONFIG_SOC_SERIES_STM32H7X || CONFIG_SOC_SERIES_STM32H7RSX)
+		#define MCO1_SOURCE	LL_RCC_MCO1SOURCE_PLL1QCLK
+	#else
+		#error "PLLQCLK is not a valid clock source on your SOC"
+	#endif
+#elif DT_PROP(DT_NODELABEL(mco1), source) == STM32_SRC_PLLCLK_DIV2
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_PLLCLK_DIV_2
+#elif DT_PROP(DT_NODELABEL(mco1), source) == STM32_SRC_PLL2CLK
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_PLL2CLK
+#elif DT_PROP(DT_NODELABEL(mco1), source) == STM32_SRC_PLLI2SCLK
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_PLLI2SCLK
+#elif DT_PROP(DT_NODELABEL(mco1), source) == STM32_SRC_PLLI2SCLK_DIV2
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_PLLI2SCLK_DIV2
+#elif DT_PROP(DT_NODELABEL(mco1), source) == STM32_SRC_SYSCLK
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_SYSCLK
+#else
+#error "MCO1 source not a valid clock source"
+#endif
+#endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(mco1), okay) */
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(mco2), okay)
+#if DT_PROP(DT_NODELABEL(mco2), source) == NO_SEL
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_SYSCLK
+#elif DT_PROP(DT_NODELABEL(mco2), source) == STM32_SRC_PLLI2S
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_PLLI2S
+#elif DT_PROP(DT_NODELABEL(mco2), source) == STM32_SRC_HSE
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_HSE
+#elif DT_PROP(DT_NODELABEL(mco2), source) == STM32_SRC_LSI
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_LSI
+#elif DT_PROP(DT_NODELABEL(mco2), source) == STM32_SRC_CSI
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_CSI
+#elif DT_PROP(DT_NODELABEL(mco2), source) == STM32_SRC_PLLCLK
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_PLLCLK
+#elif DT_PROP(DT_NODELABEL(mco2), source) == STM32_SRC_PLLPCLK
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_PLL1PCLK
+#elif DT_PROP(DT_NODELABEL(mco2), source) == STM32_SRC_PLL2PCLK
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_PLL2PCLK
+#else
+#error "MCO2 source not a valid clock source"
+#endif
+#endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(mco2), okay) */
+
+#define fn_mco_prescaler(i, v) LL_RCC_MCO ## i ## _DIV_ ## v
+#define mco_prescaler(i, v) fn_mco_prescaler(i, v)
+
+#define fn_mco_source(i) MCO ## i ## _SOURCE
+#define mco_source(i) fn_mco_source(i)
```

Comment:
See previous comment:

```suggestion
#define fn_mco_prescaler(i, v) LL_RCC_MCO ## i ## _DIV_ ## v
#define mco_prescaler(inst) fn_mco_prescaler(DT_INST_REG_ADDR(inst),        \
                                             DT_INST_PROP(inst, prescaler))

#define fn_mco_source(i) MCO ## i ## _SOURCE
#define mco_source(inst) fn_mco_source(DT_INST_REG_ADDR(inst))
```

Reply:
Possible that another layer is needed for proper expansion; I think it's fine to do so.